mars/dataframe/datasource/core.py

```diff
@@ -58,9 +58,7 @@ def _tile_head(cls, op: "HeadOptimizedDataSource"):
         # execute first chunk
         yield chunks[:1]
 
-        ctx = get_context()
-        chunk_shape = ctx.get_chunks_meta([chunks[0].key], fields=["shape"])[0]["shape"]
-
+        chunk_shape = chunks[0].shape
```

Comment:
I'm a little confused why it is possible to replace `ctx.get_chunks_meta` here.

Reply:
Before getting the shape of chunk[0], the tiling of `HeadOptimizedDataSource` yield chunks[:1] so that the chunk[0] has been executed in a stage. The task processor updates the meta of stage output _(the chunk[0] in this case)_ at the end of stage, so we can get  the shape from the chunk instead of meta service.